lib/jxl/dec_patch_dictionary.cc

```diff
@@ -35,7 +35,7 @@
 
 namespace jxl {
 
-constexpr int kMaxPatches = 1 << 24;
+constexpr size_t kMaxPatches = 1 << 24;
```

Comment:
In the profile amendment, the actual limit is `min(1<<24, width*height/16))`, but I guess we are free to not do the size-dependent thing and decode some bitstreams that are technically using too many patches.